sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -355,14 +354,12 @@ public <T> T findById(String containerName, Object id, Class<T> domainType) {
         CosmosEntityInformation<?, ?> cosmosEntityInformation = CosmosEntityInformation.getInstance(domainType);
         String containerPartitionKey = cosmosEntityInformation.getPartitionKeyFieldName();
         if ("id".equals(containerPartitionKey) && id != null) {
-            try {
-                return findById(id, domainType, new PartitionKey(CosmosUtils.getStringIDValue(id)));
-            } catch (CosmosAccessException e) {
-                LOGGER.warn(e.getMessage());
-                return null;
-            }
+            return findById(id, domainType, new PartitionKey(id));
+        }
+        if (!this.pointReadWarningLogged) {
+            LOGGER.warn("The partitionKey is not id!! Consider using findById(ID id, PartitionKey partitionKey) instead. See https://aka.ms/PointReadsInSpring for more info.");
```

Comment:
Please change the error message here to inlcude at least 1 sentence for the motivation like.

The partitionKey is not id!! Consider using findById(ID id, PartitionKey partitionKey) instead to avoid the need for using a cross partition query which results in higher latency and cost than necessary. See https://aka.ms/PointReadsInSpring for more info.